source/JsMaterialX/JsMaterialXFormat/post.js

```diff
@@ -0,0 +1,448 @@
+                nodeFs.readFile(filePath, "utf8", function (err, data) {
+                    if (err) {
+                        if (++i < searchPaths.length) {
+                            resolve(loadHandler());
+                        } else {
+                            reject(new Error("MaterialX file not found: " + fileName));
+                        }
+                    } else {
+                        var parsedPath = nodePath.parse(filePath);
+                        var path = filePath.substring(parsedPath.root.length);
+                        path = path.replace(new RegExp(pathSep, "g"), "/");
+                        resolve({
+                            data: data,         // file content
+                            filePath: path,     // file path relative to root
+                            fullPath: filePath, // the full absolute path
+
+                        });
+                    }
+                });
+            });
+        }
+
+        return loadHandler();
+    }
+
+    // Concatenate searchPath, MATERIALX_SEARCH_PATH_ENV_VAR and './' and clean up the list.
+    function prepareSearchPaths(searchPath) {
+        var newSearchPath = searchPath.concat(PATH_LIST_SEPARATOR,
+            Module.getEnviron(Module.MATERIALX_SEARCH_PATH_ENV_VAR));
+        var searchPaths = ["." + pathSep].concat(newSearchPath.split(PATH_LIST_SEPARATOR));
+        var i = searchPaths.length;
+        while (i--) {
+            if (searchPaths[i].trim() === "") {
+                searchPaths.splice(i, 1);
+            }
+        }
+        return removeDuplicates(searchPaths);
+    }
+
+    // Modify absolute paths so that they work in the WASM file system.
+    // If paths is an array, the elements will be modified in place. If it's a string, a new string will be returned.
+    function makeWasmAbsolute(paths, wasmRootFolder) {
+        var pathList = paths;
+        if (typeof paths === 'string') {
+            pathList = paths.split(PATH_LIST_SEPARATOR);
+        }
+        // Rewrite absolute paths to be absolute in the WASM FS.
+        for (var i = 0; i < pathList.length; ++i) {
+            var path = pathList[i];
+            if (ENVIRONMENT_IS_NODE) {
+                if (nodePath.isAbsolute(path)) {
+                    var parsed = nodePath.parse(path);
+                    path = wasmRootFolder + "/" + parsed.dir.substring(parsed.root.length);
+                }
+                path.replace(new RegExp(pathSep, "g"), "/");
+            } else if (ENVIRONMENT_IS_WEB) {
+                var link = document.createElement("a");
+                link.href = path;
+                if (link.origin + link.pathname + link.search + link.hash === path) {
+                    path = wasmRootFolder + link.pathname;
+                }
+            } else {
+                throw new Error("Unknown environment!");
+            }
+            pathList[i] = path;
+        }
+        if (typeof paths === 'string') {
```

Comment:
not fan of the return value if string otherwise look into the object. In Threejs they return the modified object aswell. I would say you always return something